### PR TITLE
[PBA-4524] Adding video player plugin sample in AdvancedPlaybackSampleApp

### PR DESCRIPTION
**Pull request by akshaydch11**

### CODE REVIEW

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoPlayerFactory.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.ooyala.android.player.MoviePlayer;
+import com.ooyala.android.player.PlayerFactory;
+import com.ooyala.android.player.StreamPlayer;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by achaudhari on 9/9/16.
+ */
```
Comment (lental) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoPlayerFactory.java:15]: remove names from all new files

Comment (lental) [reply AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoPlayerFactory.java:15]: please add a little documentation on each public class, to explain what it is for

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+ * Created by achaudhari on 9/12/16.
+ */
+public class SampleVideoStreamPlayer extends StreamPlayer {
+
+    private static final String TAG = SampleVideoStreamPlayer.class.getSimpleName();
+
+    private int DURATION;
+    private final int REFRESH_RATE = 250;
+
+    private Timer _timer;
```
Comment (lental) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java]: can you remove underscores?  We have been slowly enforcing the removal of underscores for variables in android

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+        if (_parent.getDesiredState() == OoyalaPlayer.DesiredState.DESIRED_PLAY) {
+          setState(OoyalaPlayer.State.PLAYING);
+        } else {
+          setState(OoyalaPlayer.State.READY);
+        }
+
+    }
+
+    @Override
+    protected void startPlayheadTimer() {
```
Comment (lental) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java]: do you need to override this?

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+    public void initPlayer () {
+        _linearLayout = new LinearLayout(_parent.getLayout().getContext());
+        _linearLayout.setLayoutParams(new FrameLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT, Gravity.CENTER));
+        _linearLayout.setBackgroundColor(Color.BLACK);
+        textView = new TextView(_parent.getLayout().getContext());
+        _linearLayout.addView(textView);
+        _parent.getLayout().addView(_linearLayout);
```
Comment (lental) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java]: add a comment, that this is where you specifically add your player's view to the parent view

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        if (_parent.getDesiredState() == OoyalaPlayer.DesiredState.DESIRED_PLAY) {
+          setState(OoyalaPlayer.State.PLAYING);
+        } else {
+          setState(OoyalaPlayer.State.READY);
+        }
+
+    }
+
+    @Override
+    public void resume() {
```
Comment (brol1dev) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java]: To make this cleaner, can you remove the @override methods that aren't doing anything?

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+import com.ooyala.android.util.DebugMode;
+
+import java.util.Set;
+import java.util.Timer;
+import java.util.TimerTask;
+
+
+/**
+ *  This is a new player which extends from StreamPlayer.
+ *  Extending from StreamPlayer is necessary to get it work with Ooyala SDk.
```
Comment (brol1dev) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java]: Capital K in SDk, please :)

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    public void initPlayer () {
+        // create layout/view which will contain playerview
+        linearLayout = new LinearLayout(_parent.getLayout().getContext());
+        linearLayout.setLayoutParams(new FrameLayout.LayoutParams(
+                ViewGroup.LayoutParams.MATCH_PARENT,
+                ViewGroup.LayoutParams.MATCH_PARENT, Gravity.CENTER));
+        linearLayout.setBackgroundColor(Color.BLACK);
+        textView = new TextView(_parent.getLayout().getContext());
+        linearLayout.addView(textView);
+        // Add newly created playerView/ to parent which is Ooyala player
```
Comment (brol1dev) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java]: remove forward slash after 'playerView/'

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+    public void reset() {
+        // TODO Auto-generated method stub
+    }
+
+    @Override
+    public void seekToTime(int timeInMillis) {
+
+    }
+
+    private void refresh() {
```
Comment (brol1dev) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/utils/SampleVideoStreamPlayer.java:129]: Maybe add comments to this method saying that it simulates playback.

File: AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SampleVideoPlayerActivity.java [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+        setContentView(R.layout.player_simple_layout);
+
+        EMBED = getIntent().getExtras().getString("embed_code");
+        PCODE = getIntent().getExtras().getString("pcode");
+        DOMAIN = getIntent().getExtras().getString("domain");
+
+        OoyalaPlayerLayout playerLayout = (OoyalaPlayerLayout) findViewById(R.id.ooyalaPlayer);
+        player = new OoyalaPlayer(PCODE, new PlayerDomain(DOMAIN));
+        playerLayoutController = new OoyalaPlayerLayoutController(playerLayout, player);
+        // Below step will make sure that player is being chose each time
```
Comment (brol1dev) [inline AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SampleVideoPlayerActivity.java]: chosen instead of 'chose'

**Review by brol1dev — APPROVED** [review]
Check my comments and you're good to go for me :)

**Comment by brol1dev** [comment]
@lental, do you have anything else to say? If not we can merge it.

**Comment by lental** [comment]
go for it :)

**Pull request merged**